test_mypy_error fails with mypy >= 0.650, < 0.700

```
py34-pytest2.9-mypy0.650 create: /Users/davidmtucker/Projects/pytest-mypy/.tox/py34-pytest2.9-mypy0.650
py34-pytest2.9-mypy0.650 installdeps: pytest ~= 2.9.0, mypy == 0.650
py34-pytest2.9-mypy0.650 inst: /Users/davidmtucker/Projects/pytest-mypy/.tox/.tmp/package/1/pytest-mypy-0.3.2.zip
py34-pytest2.9-mypy0.650 installed: DEPRECATION: Python 3.4 support has been deprecated. pip 19.1 will be the last one supporting it. Please upgrade your Python as Python 3.4 won't be maintained after March 2019 (cf PEP 429).,mypy==0.650,mypy-extensions==0.4.1,py==1.8.0,pytest==2.9.2,pytest-mypy==0.3.2,typed-ast==1.1.2,typing==3.6.6
py34-pytest2.9-mypy0.650 run-test-pre: PYTHONHASHSEED='2665214780'
py34-pytest2.9-mypy0.650 runtests: commands[0] | py.test tests
========================================================================= test session starts =========================================================================
platform darwin -- Python 3.4.10, pytest-2.9.2, py-1.8.0, pluggy-0.3.1
rootdir: /Users/davidmtucker/Projects/pytest-mypy, inifile: 
plugins: mypy-0.3.2
collected 2 items 

tests/test_pytest_mypy.py .F

============================================================================== FAILURES ===============================================================================
___________________________________________________________________________ test_mypy_error ___________________________________________________________________________

testdir = <Testdir local('/private/var/folders/th/55zpp9zj6vddgfjy3n3p9h0w0000gn/T/pytest-of-davidmtucker/pytest-16/testdir/test_mypy_error0')>

    def test_mypy_error(testdir):
        testdir.makepyfile('''
            def myfunc(x: int) -> str:
                return x * 2
    
            def test_myfunc():
                assert myfunc(12)
        ''')
    
        result = testdir.runpytest('--mypy', '-v')
    
        result.stdout.fnmatch_lines([
>           'test_mypy_error.py:2: error: Incompatible return value*',
        ])
E       Failed: remains unmatched: 'test_mypy_error.py:2: error: Incompatible return value*', see stderr

/Users/davidmtucker/Projects/pytest-mypy/tests/test_pytest_mypy.py:27: Failed
------------------------------------------------------------------------ Captured stdout call -------------------------------------------------------------------------
============================= test session starts ==============================
platform darwin -- Python 3.4.10, pytest-2.9.2, py-1.8.0, pluggy-0.3.1 -- /Users/davidmtucker/Projects/pytest-mypy/.tox/py34-pytest2.9-mypy0.650/bin/python
cachedir: .cache
rootdir: /private/var/folders/th/55zpp9zj6vddgfjy3n3p9h0w0000gn/T/pytest-of-davidmtucker/pytest-16/testdir/test_mypy_error0, inifile: 
plugins: mypy-0.3.2
collecting ... collected 2 items

test_mypy_error.py PASSED
test_mypy_error.py::test_myfunc PASSED

=========================== 2 passed in 0.31 seconds ===========================
------------------------------------------------------------------------ Captured stderr call -------------------------------------------------------------------------
nomatch: 'test_mypy_error.py:2: error: Incompatible return value*'
    and: '============================= test session starts =============================='
    and: 'platform darwin -- Python 3.4.10, pytest-2.9.2, py-1.8.0, pluggy-0.3.1 -- /Users/davidmtucker/Projects/pytest-mypy/.tox/py34-pytest2.9-mypy0.650/bin/python'
    and: 'cachedir: .cache'
    and: 'rootdir: /private/var/folders/th/55zpp9zj6vddgfjy3n3p9h0w0000gn/T/pytest-of-davidmtucker/pytest-16/testdir/test_mypy_error0, inifile: '
    and: 'plugins: mypy-0.3.2'
    and: 'collecting ... collected 2 items'
    and: ''
    and: 'test_mypy_error.py PASSED'
    and: 'test_mypy_error.py::test_myfunc PASSED'
    and: ''
    and: '=========================== 2 passed in 0.31 seconds ==========================='
    and: ''
================================================================= 1 failed, 1 passed in 3.38 seconds ==================================================================
ERROR: InvocationError for command '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34-pytest2.9-mypy0.650/bin/py.test tests' (exited with code 1)
```

This is specific to Python 3.4:
``` diff
diff --git a/pytest_mypy.py b/pytest_mypy.py
index d98bc59..97ef49f 100644
--- a/pytest_mypy.py
+++ b/pytest_mypy.py
@@ -66,10 +66,13 @@ class MypyItem(pytest.Item, pytest.File):
 
         mypy_argv += self.mypy_config
 
-        stdout, _, _ = mypy.api.run(args=mypy_argv)
+        stdout, stderr, status = mypy.api.run(args=mypy_argv)
 
+        if stderr:
+            raise MypyError(stderr)
         if stdout:
             raise MypyError(stdout)
+        assert status == 0
 
     def repr_failure(self, excinfo):
         """
```
```
___________________________________________________________________________ test_mypy_error ___________________________________________________________________________

testdir = <Testdir local('/private/var/folders/th/55zpp9zj6vddgfjy3n3p9h0w0000gn/T/pytest-of-davidmtucker/pytest-1001/test_mypy_error0')>

    def test_mypy_error(testdir):
        testdir.makepyfile('''
            def myfunc(x: int) -> str:
                return x * 2
    
            def test_myfunc():
                assert myfunc(12)
        ''')
    
        result = testdir.runpytest('--mypy', '-v')
    
        result.stdout.fnmatch_lines([
>           'test_mypy_error.py:2: error: Incompatible return value*',
        ])
E       Failed: nomatch: 'test_mypy_error.py:2: error: Incompatible return value*'
E           and: '============================= test session starts =============================='
E           and: 'platform darwin -- Python 3.4.10, pytest-4.5.0, py-1.8.0, pluggy-0.11.0 -- /Users/davidmtucker/Projects/pytest-mypy/.tox/py34/bin/python'
E           and: 'cachedir: .pytest_cache'
E           and: 'rootdir: /private/var/folders/th/55zpp9zj6vddgfjy3n3p9h0w0000gn/T/pytest-of-davidmtucker/pytest-1001/test_mypy_error0'
E           and: 'plugins: mypy-0.3.2'
E           and: 'collecting ... collected 2 items'
E           and: ''
E           and: 'test_mypy_error.py FAILED                                                [ 50%]'
E           and: 'test_mypy_error.py::test_myfunc PASSED                                   [100%]'
E           and: ''
E           and: '=================================== FAILURES ==================================='
E           and: '___________________________ mypy test_mypy_error.py ____________________________'
E           and: '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/hooks.py:289: in __call__'
E           and: '    return self._hookexec(self, self.get_hookimpls(), kwargs)'
E           and: '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/manager.py:68: in _hookexec'
E           and: '    return self._inner_hookexec(hook, methods, kwargs)'
E           and: '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/_tracing.py:80: in __call__'
E           and: '    return outcome.get_result()'
E           and: '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/_tracing.py:78: in <lambda>'
E           and: '    outcome = _Result.from_call(lambda: self.oldcall(hook, hook_impls, kwargs))'
E           and: '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/manager.py:62: in <lambda>'
E           and: '    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,'
E           and: '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/_pytest/runner.py:122: in pytest_runtest_call'
E           and: '    item.runtest()'
E           and: '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pytest_mypy.py:75: in runtest'
E           and: '    assert status == 0'
E           and: "E   AssertionError: assert 'Running mypy... 3.5 or newer' == 0"
E           and: "E     -'Running mypy with Python 3.4 or lower is not supported; please upgrade to 3.5 or newer'"
E           and: 'E     +0'
E           and: '=============================== warnings summary ==============================='
E           and: '/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/_pytest/mark/structures.py:324'
E           and: '  /Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/_pytest/mark/structures.py:324: PytestUnknownMarkWarning: Unknown pytest.mark.mypy - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html'
E           and: '    PytestUnknownMarkWarning,'
E           and: ''
E           and: '-- Docs: https://docs.pytest.org/en/latest/warnings.html'
E           and: '================ 1 failed, 1 passed, 1 warnings in 0.12 seconds ================'
E           and: ''
E       remains unmatched: 'test_mypy_error.py:2: error: Incompatible return value*'

/Users/davidmtucker/Projects/pytest-mypy/tests/test_pytest_mypy.py:27: Failed
------------------------------------------------------------------------ Captured stdout call -------------------------------------------------------------------------
============================= test session starts ==============================
platform darwin -- Python 3.4.10, pytest-4.5.0, py-1.8.0, pluggy-0.11.0 -- /Users/davidmtucker/Projects/pytest-mypy/.tox/py34/bin/python
cachedir: .pytest_cache
rootdir: /private/var/folders/th/55zpp9zj6vddgfjy3n3p9h0w0000gn/T/pytest-of-davidmtucker/pytest-1001/test_mypy_error0
plugins: mypy-0.3.2
collecting ... collected 2 items

test_mypy_error.py FAILED                                                [ 50%]
test_mypy_error.py::test_myfunc PASSED                                   [100%]

=================================== FAILURES ===================================
___________________________ mypy test_mypy_error.py ____________________________
/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/hooks.py:289: in __call__
    return self._hookexec(self, self.get_hookimpls(), kwargs)
/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/manager.py:68: in _hookexec
    return self._inner_hookexec(hook, methods, kwargs)
/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/_tracing.py:80: in __call__
    return outcome.get_result()
/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/_tracing.py:78: in <lambda>
    outcome = _Result.from_call(lambda: self.oldcall(hook, hook_impls, kwargs))
/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pluggy/manager.py:62: in <lambda>
    firstresult=hook.spec.opts.get("firstresult") if hook.spec else False,
/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/_pytest/runner.py:122: in pytest_runtest_call
    item.runtest()
/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/pytest_mypy.py:75: in runtest
    assert status == 0
E   AssertionError: assert 'Running mypy... 3.5 or newer' == 0
E     -'Running mypy with Python 3.4 or lower is not supported; please upgrade to 3.5 or newer'
E     +0
=============================== warnings summary ===============================
/Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/_pytest/mark/structures.py:324
  /Users/davidmtucker/Projects/pytest-mypy/.tox/py34/lib/python3.4/site-packages/_pytest/mark/structures.py:324: PytestUnknownMarkWarning: Unknown pytest.mark.mypy - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/latest/mark.html
    PytestUnknownMarkWarning,

-- Docs: https://docs.pytest.org/en/latest/warnings.html
================ 1 failed, 1 passed, 1 warnings in 0.12 seconds ================
```

--- a/setup.py
+++ b/setup.py
@@ -23,7 +23,11 @@ setup(
     description='Mypy static type checker plugin for Pytest',
     long_description=read('README.rst'),
     py_modules=['pytest_mypy'],
-    install_requires=['pytest>=2.9.2', 'mypy~=0.570'],
+    install_requires=[
+        'pytest>=2.9.2',
+        'mypy>=0.570,<0.650; python_version<"3.5"',
+        'mypy~=0.570; python_version>="3.5"',
+    ],
     classifiers=[
         'Development Status :: 4 - Beta',
         'Framework :: Pytest',

